client/python/conductor/FrinxConductorWrapper.py

```diff
@@ -0,0 +1,162 @@
+import copy
+import json
+import threading
+import time
+import traceback
+import logging
+from datetime import datetime
+from threading import Thread
+
+import requests
+from conductor.ConductorWorker import ConductorWorker
+
+log = logging.getLogger(__name__)
+
+DEFAULT_TASK_DEFINITION = {
+    "name": "",
+    "retryCount": 0,
+    "ownerEmail": "example@example.com",
+    "timeoutSeconds": 60,
+    "timeoutPolicy": "TIME_OUT_WF",
+    "retryLogic": "FIXED",
+    "retryDelaySeconds": 0,
+    "responseTimeoutSeconds": 10
+}
+
+
+class FrinxConductorWrapper(ConductorWorker):
+    """
+    Adds:
+    - Exception handling: auto failing tasks on exception
+    - Task registration function
+    - Batch polling for tasks instead of simple polling
+    - Dedicated queue scanning thread to only perform polling if there are tasks waiting in queue
+    """
+
+    def __init__(self, server_url, thread_count, polling_interval, worker_id=None, headers=None):
+        # Synchronizes access to self.queues by producer thread (in read_queue) and consumer threads (in tasks_in_queue)
+        self.lock = threading.Lock()
+        self.queues = {}
+        self.conductor_task_url = server_url + "/metadata/taskdefs"
+        self.headers= headers
+        super().__init__(server_url, thread_count, polling_interval, worker_id, headers)
+
+    def start_queue_polling(self):
+        log.debug("Starting queue polling thread")
+        thread = Thread(target=self.read_queues)
+        thread.daemon = True
+        thread.start()
+
+    def read_queues(self):
+        failCount = 0
+        log.debug("Queue polling thread started")
+        while True:
+            try:
+                time.sleep(float(self.polling_interval))
+                self.lock.acquire()
+                queuesTemp = self.taskClient.getTasksInQueue("all")
+                log.debug(f'Queues polled: {queuesTemp}')
```

Comment:
I know you did not do it this way (... actually I did)

but it should be changed according to Daniel's comment to my initial change:

You should not use f string in logging, because it is evaluated in runtime.
You should use some lazy evaluation: log.info("Queues polled: %s", queuesTemp)
Please unify across whole file.